_test/template.response.test.php

```diff
@@ -0,0 +1,328 @@
+        $this->assertEquals($expectedLegend, $legend, 'Expected legend should be equal to the translation set in the english translation file and should contain one item when one page has no title set.');
+    }
+
+    public function test_response_has_one_element_when_two_pages_has_no_title_set()
+    {
+        $pageHasNoTitle = ['ihavenotitle1', 'ihavenotitle2'];
+        Helper::savePages($pageHasNoTitle);
+
+        ob_start();
+
+        $adminPlugin = new admin_plugin_findologicxmlexport();
+        $adminPlugin->html();
+        $output = ob_get_clean();
+        $dom = new \DOMDocument('1.0', 'UTF-8');
+
+
+        $internalErrors = libxml_use_internal_errors(true);
+        $dom->loadHTML($output);
+        libxml_use_internal_errors($internalErrors);
+
+        $header = $dom->getElementsByTagName('h1')[0];
+        $header = $header->nodeValue;
+
+        $expectedHeader = 'FINDOLOGIC XML Export Plugin';
+
+        $legend = $dom->getElementsByTagName('legend')[0];
+        $legend = trim($legend->nodeValue);
+
+        $expectedLegend = 'Pages without title (2)';
+
+        $this->assertEquals($expectedHeader, $header, 'Expected header should be equal to the translation set in the english translation file.');
+        $this->assertEquals($expectedLegend, $legend, 'Expected legend should be equal to the translation set in the english translation file and should contain one item when one page has no title set.');
+    }
+
+    public function test_response_have_multiple_elements_when_six_or_more_pages_has_no_title_set()
+    {
+        // Save pages
+        $pagesHaveNoTitle = ['noootitle324', 'noootitle1784', 'noootitle1203', 'noootitle356', 'noootitle1337', 'noootitle1338'];
+        foreach ($pagesHaveNoTitle as $pageHasNoTitle) {
+            Helper::savePages([$pageHasNoTitle]);
+        }
+
+        ob_start();
+        $adminPlugin = new admin_plugin_findologicxmlexport();
+        $adminPlugin->html();
+        $output = ob_get_clean();
+        $dom = new \DOMDocument('1.0', 'UTF-8');
+
+        $internalErrors = libxml_use_internal_errors(true);
+        $dom->loadHTML($output);
+        libxml_use_internal_errors($internalErrors);
+
+        $finder = new DomXPath($dom);
+
+        $pageFields = ['page-id', 'page-url', 'page-author', 'page-last-edited'];
+
+        foreach ($pageFields as $key => $pageField) {
+
+            $nodes = $finder->query("//td[contains(@class, '$pageField')]");
+
+            for ($i = 0; $i < 5; $i++) {
+                $pageWithoutTitle[$pageField][] = trim($nodes->item($i)->nodeValue);
+                sort($pageWithoutTitle[$pageField]);
+            }
+        }
+
+        // page-id
+        $expectedPageWithoutTitle['page-id'] = [
+            'noootitle1338',
+            'noootitle1337',
+            'noootitle356',
+            'noootitle1203',
+            'noootitle1784'
+        ];
+        sort($expectedPageWithoutTitle['page-id']);
```

Comment:
I assume you want to sort the elements of `$expectedPageWithoutTitle` by `page-id`. I don't think this does what you want:

```
$ php -a
Interactive mode enabled

php > $a = [['key' => 'valueB'], ['key' => 'valueA']];
php > var_dump($a);
php shell code:1:
array(2) {
  [0] =>
  array(1) {
    'key' =>
    string(6) "valueB"
  }
  [1] =>
  array(1) {
    'key' =>
    string(6) "valueA"
  }
}
php > sort($a['key']);
PHP Warning:  sort() expects parameter 1 to be array, null given in php shell code on line 1
PHP Stack trace:
PHP   1. {main}() php shell code:0
PHP   2. sort() php shell code:1
```

Additionally, this should throw a warning which should make your tests fail. Please make sure that they fail.

Reply:
:white_check_mark: As we discussed together, this test works fine